Open Judge System/LocalWorker/OJS.LocalWorker/OjsSubmissionProcessingStrategy.cs

```diff
@@ -170,6 +174,29 @@ protected override void ProcessTestsExecutionResult(IExecutionResult<TestResult>
             this.UpdateResults();
         }
 
+        private void AssignWorkerType()
+        {
+            var currentSubmission = this.submissionsData.GetById(this.submissionForProcessing.SubmissionId);
+            this.submissionForProcessing.WorkerType = currentSubmission.WorkerType;
+
+            if (this.submissionForProcessing.WorkerType == WorkerType.None)
+            {
+                var contestWorkerType = currentSubmission.Problem.ProblemGroup.Contest.DefaultWorkerType;
+                var strategyDetailsWorkerType = currentSubmission.Problem
+                    .ProblemSubmissionTypeExecutionDetails
```

Comment:
This is very inefficient (+ you will make several unnecessary requests due to lazy loading). We can have the correct WorkerType prepared beforehand in the `ExecutionDetails`, because upon Submit we should already have all the data needed.

Reply:
Fixed.